src/suggestions/index.jsx

```diff
@@ -24,6 +23,13 @@ const KEY_DOWN = 40;
  */
 const suggestionMatcher = /(?:^|\s)@([^\s]*)$/i;
 
+/**
+ * This pattern looks for a query
+ *
+ * @type {RegExp} matches @query
+ */
+const queryMatcher = query => new RegExp(`^${query}| ${query}`, 'i'); // start of string, or preceded by a space
```

Comment:
not sure what `query` should be but this will treat it as a RegExp object. that is, if `query === '[abc]'` then it will match only a single-letter string of `a`, `b`, or `c`.

lodash has a useful function to escape strings we inject into a RegExp object: `lodash.escapeRegExp( string )` (no surprise in the name, lol)

might be useful to reflect on what our intent here is